src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
-    this.includeReadme = options.readme ?? true;
+    // normally the assemblies are located in subdirectories
+    // of the root package dir (i.e ./node_modules)
+    const assembliesDir = options?.assembliesDir ?? root;
+
+    const manifest = JSON.parse(await fs.readFile(manifestPath, 'utf-8'));
+    return Documentation.forAssembly(manifest.name, assembliesDir);
+  }
+
+  /**
+   * Create a `Documentation` object for a specific assembly that resides in a directory of assemblies.
+   * The directory will comprise the entire type-system available to the generartion process.
+   */
+  public static async forAssembly(assemblyName: string, assembliesDir: string, options?: DocumentationOptions): Promise<Documentation> {
+    return withTempDir(async (workdir: string) => {
+
+      // always better not to operate on an externally provided directory
+      await fs.copy(assembliesDir, workdir);
+
+      let transpile, language;
+
+      switch (options?.language ?? 'ts') {
+        case 'python':
+          language = TargetLanguage.PYTHON;
+          transpile = new PythonTranspile();
+          break;
+        case 'ts':
+          transpile = new TypeScriptTranspile();
+          break;
+        default:
+          throw new Error(`Unsupported language: ${options?.language}`);
+      }
+      const assembly = await createAssembly(assemblyName, workdir, options?.loose ?? true, language);
+      return new Documentation(assembly, transpile);
+    });
+
+  }
+
+  private constructor(
+    private readonly assembly: reflect.Assembly,
+    private readonly transpile: Transpile) {
   }
 
   /**
    * Generate markdown.
    */
-  public render(): Markdown {
+  public render(options?: RenderOptions): Markdown {
+    const submodule = options?.submodule ? this.findSubmodule(this.assembly, options.submodule) : undefined;
     const documentation = new Markdown();
 
-    if (this.includeReadme) {
-      const readme = new Readme(this.transpile, this.assembly, this.submodule);
+    if (options?.readme ?? true) {
+      const readme = new Readme(this.transpile, this.assembly, submodule);
       documentation.section(readme.render());
     }
-    if (this.includeApiReference) {
-      const apiReference = new ApiReference(
-        this.transpile,
-        this.assembly,
-        this.submodule,
-      );
+
+    if (options?.apiReference ?? true) {
+      const apiReference = new ApiReference(this.transpile, this.assembly, submodule);
       documentation.section(apiReference.render());
     }
 
     return documentation;
   }
 
   /**
-   * Lookup a submodule by a submodule name given by the caller.
+   * Lookup a submodule by a submodule name.
    */
-  private findSubmodule(submoduleName: string): reflect.Submodule {
-    const submodules = this.assembly.submodules.filter(
-      (s) => s.name === submoduleName,
+  private findSubmodule(assembly: reflect.Assembly, submodule?: string): reflect.Submodule {
```

Comment:
It makes no sense to me that the `submodule` parameter of a `findSubmodule` method is optional... How do you go and find a submodule by name if you're not given its name?

Reply:
Yeap, this was a leftover, I actually check for `null` before i invoke this method, so not needed